packages/nimble-components/src/table/models/expansion-manager.ts

```diff
@@ -0,0 +1,76 @@
+import type { Row as TanStackRow } from '@tanstack/table-core';
+import type { TableNode, TableRecord } from '../types';
+
+/**
+ * Manages the expanded/collapsed state of group rows and parent rows in the table. We must track
```

Comment:
Feel free to take/reject any number that you think make sense.

_Manages the expanded/collapsed state of rows in the table._

_We must track the expansion state separately from TanStack because:_
_1. TanStack does not support having a different initial expansion state per row unless explicitly specified for each row by ID. This causes problems in the nimble-table because we need to have a different initial expansion state for group rows, parent rows, and parent rows with lazy loaded children._
_2. TanStack does not remove entries from its expanded state when those rows are no longer present in the data. This is not ideal because the object maintaining the expansion state can grow unbounded._